js/bitopro.js

```diff
@@ -0,0 +1,1335 @@
+        }
+        await this.loadMarkets ();
+        const currency = this.safeCurrency (code);
+        const request = {
+            'serial': id,
+            'currency': currency['id'],
+        };
+        const response = await this.privateGetWalletWithdrawCurrencySerial (this.extend (request, params));
+        const result = this.safeValue (response, 'data', {});
+        //
+        //     {
+        //         "data":{
+        //             "serial":"20220215BW14069838",
+        //             "address":"TKrwMaZaGiAvtXCFT41xHuusNcs4LPWS7w",
+        //             "amount":"8.00000000",
+        //             "fee":"2.00000000",
+        //             "total":"10.00000000",
+        //             "status":"COMPLETE",
+        //             "txid":"50bf250c71a582f40cf699fb58bab978437ea9bdf7259ff8072e669aab30c32b",
+        //             "protocol":"TRX",
+        //             "id":"9925310345",
+        //             "timestamp":"1644907716044"
+        //         }
+        //     }
+        //
+        return this.parseTransaction (result, currency);
+    }
+
+    async withdraw (code, amount, address, tag = undefined, params = {}) {
+        [ tag, params ] = this.handleWithdrawTagAndParams (tag, params);
+        await this.loadMarkets ();
+        this.checkAddress (address);
+        const currency = this.currency (code);
+        const request = {
+            'currency': currency['id'],
+            'amount': this.numberToString (amount),
+            'address': address,
+            // 'protocol': '',
+        };
+        if (tag !== undefined) {
+            request['message'] = tag;
+        }
+        const response = await this.privatePostWalletWithdrawCurrency (this.extend (request, params));
+        const result = this.safeValue (response, 'data', {});
+        //
+        //     {
+        //         "data":{
+        //             "serial":"20220215BW14069838",
+        //             "currency":"USDT",
+        //             "protocol":"TRX",
+        //             "address":"TKrwMaZaGiAvtXCFT41xHuusNcs4LPWS7w",
+        //             "amount":"8",
+        //             "fee":"2",
+        //             "total":"10"
+        //         }
+        //     }
+        //
+        return this.parseTransaction (result, currency);
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        let url = '/' + this.implodeParams (path, params);
+        const query = this.omit (params, this.extractParams (path));
+        if (headers === undefined) {
+            headers = {};
+        }
+        headers['X-BITOPRO-API'] = 'ccxt';
+        if (api === 'private') {
+            this.checkRequiredCredentials ();
+            if (method === 'POST') {
+                body = this.json (params);
+                const payload = this.stringToBase64 (body);
+                const signature = this.hmac (payload, this.secret, 'sha384');
```

Comment:
we need to change this line to `this.hmac (this.encode (payload), this.encode (this.secret), 'sha384)`

Reply:
It work with: `this.hmac (payload, this.encode (this.secret), 'sha384')`, I think the payload is byte.